packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthCognito.kt

```diff
@@ -223,17 +229,46 @@ public class AuthCognito : FlutterPlugin, ActivityAware, MethodCallHandler, Plug
   }
 
   private fun onSignIn (@NonNull flutterResult: Result, @NonNull request: HashMap<String, *>) {
-      try {
-        FlutterSignInRequest.validate(request)
-        var req = FlutterSignInRequest(request)
-        Amplify.Auth.signIn(
-                req.username,
-                req.password,
-                { result -> prepareSignInResult(flutterResult, result) },
-                { error -> errorHandler.handleAuthError(flutterResult, error)}
-        );
-      } catch (e: Exception) {
-        errorHandler.prepareGenericException(flutterResult, e)
+
+      runBlocking {
+        suspend fun fetchAuthSession(): AuthSession {
+          return suspendCoroutine { continuation ->
+            Amplify.Auth.fetchAuthSession(
+                    { continuation.resume(it) },
+                    { continuation.resumeWithException(it) }
+            )
+          }
+        }
+
+        suspend fun signIn(): AuthSignInResult {
+          FlutterSignInRequest.validate(request)
+          var req = FlutterSignInRequest(request)
+          return suspendCoroutine { continuation ->
+            Amplify.Auth.signIn(req.username, req.password,
+                    { continuation.resume(it) },
+                    { continuation.resumeWithException(it) }
+            )
+          }
+        }
```

Comment:
I'd pull these out of the `runBlocking`, and make them ordinary functions inside of your `AuthCognito` class. Well, and mark them `private`.

Reply:
Done (I made them private suspend functions, of course)